store/store.go

```diff
@@ -1159,73 +1171,97 @@ var download = func(name, sha3_384, downloadURL string, user *auth.UserState, s
 }
 
 // downloadDelta downloads the delta for the preferred format, returning the path.
-func (s *Store) downloadDelta(name string, downloadDir string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (string, error) {
+func (s *Store) downloadDelta(deltaName string, downloadInfo *snap.DownloadInfo, w io.Writer, pbar progress.Meter, user *auth.UserState) error {
 
 	if len(downloadInfo.Deltas) != 1 {
-		return "", errors.New("store returned more than one download delta")
+		return errors.New("store returned more than one download delta")
 	}
 
 	deltaInfo := downloadInfo.Deltas[0]
 
 	if deltaInfo.Format != s.deltaFormat {
-		return "", fmt.Errorf("store returned a download delta with the wrong format (%q instead of the configured %s format)", deltaInfo.Format, s.deltaFormat)
+		return fmt.Errorf("store returned a download delta with the wrong format (%q instead of the configured %s format)", deltaInfo.Format, s.deltaFormat)
```

Comment:
Done. RE the s.deltaFormat - I'd originally made it a configurable option of the store defaulting to xdelta (as it will need to be configurable eventually) - so that this option is sent to the store when requesting deltas. But of course when applying deltas I can only implemented xdelta currently. Happy to remove the configuration possibility in a separate branch if you think it's worth removing until we need it to be configurable.